### PR TITLE
Friendlier accumulator table for native compute w.r.t. naked pointers.

### PR DESCRIPTION
**Pull request by ppedrot**
Instead of unconditionally allocating the rel accu table at the process startup, we delay its initialization until the first rel accu is generated. This way we do not create naked pointers if the native compiler is never called.

This was mentioned in #13940 as a quick fix for the no-naked-pointer mode when not using native.

### CODE REVIEW

File: kernel/nativevalues.ml [hunk 1]
```diff
@@ -113,10 +113,17 @@ let mk_rel_accu i =
   mk_accu (Arel i)
 
 let rel_tbl_size = 100
-let rel_tbl = Array.init rel_tbl_size mk_rel_accu
+let rel_tbl_init = ref false
+(* Initialize the table lazily not to generate naked pointers at startup *)
+let rel_tbl = ref [||]
```
Comment (silene) [inline kernel/nativevalues.ml:118]: You do not need a reference. Just initialize the array with a dummy value, e.g., `mk_int 0`.

Comment (ejgallego) [reply kernel/nativevalues.ml]: Nice, this was a problem in the multicore branch, I dunno tho if it wouldn't be better just to wrap the value in a `Lazy.t` ?

**Review by silene — APPROVED** [review]

File: kernel/nativevalues.ml (hunk 1 above)
Comment (ejgallego) [reply kernel/nativevalues.ml]: I fear even the `mk_int 0` could be a problem in the future.

Comment (ppedrot) [reply kernel/nativevalues.ml]: If `mk_int` becomes a problem in the future, you can basically dump the native compiler in a trashbin.

Comment (silene) [reply kernel/nativevalues.ml]: > I fear even the `mk_int 0` could be a problem in the future.

That does not make sense to me. If the OCaml runtime is changed to force boxing integers on the heap, the language is just doomed.

Comment (ejgallego) [reply kernel/nativevalues.ml]: Well indeed my comment goes beyond the scope of this PR, and it has to do with what happens with native and the multicore GC.

**Comment by ppedrot** [comment]
Can we get this one in 8.15? It should make easier the experimentation of naked-pointers with a Coq released version.

**Comment by SkySkimmer** [comment]
@silene is this waiting for something?

**Comment by silene** [comment]
@coqbot: merge now

**Pull request merged**